File: app/src/main/java/com/ymovie/app/data/model/movie/MovieList.java
package com.ymovie.app.data.model.movie;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class MovieList {
    @SerializedName("page")
    @Expose
    private int page;
    @SerializedName("results")
    @Expose
    private List<Movie> movies;
    @SerializedName("total_pages")
    @Expose
    private int totalPage;
    @SerializedName("total_results")
    @Expose
    private int totalResult;

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public List<Movie> getMovies() {
        return movies;
    }

    public void setMovies(List<Movie> movies) {
        this.movies = movies;
    }

    public int getTotalPage() {
        return totalPage;
    }

    public void setTotalPage(int totalPage) {
        this.totalPage = totalPage;
    }

    public int getTotalResult() {
        return totalResult;
    }

    public void setTotalResult(int totalResult) {
        this.totalResult = totalResult;
    }
}

File: app/src/main/java/com/ymovie/app/data/model/movie/MovieList.kt
package com.ymovie.app.data.model.movie

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

const val DEFAULT_PAGE = 1
const val DEFAULT_TOTAL_PAGE = 1
const val DEFAULT_TOTAL_RESULT = 0
const val DEFAULT_HEADER = ""

data class MovieList(
    @SerializedName("page")
    @Expose
    var page: Int = DEFAULT_PAGE,
    @SerializedName("results")
    @Expose
    var movies: List<Movie>? = emptyList(),
    @SerializedName("total_pages")
    @Expose
    var totalPage: Int = DEFAULT_TOTAL_PAGE,
    @SerializedName("total_results")
    @Expose
    var totalResult: Int = DEFAULT_TOTAL_RESULT,
    var header: String = DEFAULT_HEADER
)